plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -470,6 +499,46 @@ public void setNull(PreparedStatement statement, int index)
         };
     }
 
+    public static ColumnMapping sqlServerTimestampColumnMapping(TimestampType type)
+    {
+        if (type.getPrecision() <= MAX_SHORT_PRECISION) {
+            return timestampColumnMapping(type);
+        }
+        return ColumnMapping.objectMapping(
+                type,
+                longTimestampReadFunction(type),
+                longTimestampWriteFunction());
+    }
+
+    private static ObjectReadFunction longTimestampReadFunction(TimestampType type)
+    {
+        return ObjectReadFunction.of(
+                LongTimestamp.class,
+                (resultSet, columnIndex) -> {
+                    LocalDateTime localDateTime = resultSet.getObject(columnIndex, LocalDateTime.class);
+                    Instant instant = localDateTime.atZone(UTC).toInstant();
+                    long epochMicros = instant.getEpochSecond() * MICROSECONDS_PER_SECOND + floorDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
```

Comment:
floorDiv or roundDiv?

Reply:
Anyway, if you can use LocalDateTime for SQL Server, the code here should actually be added in 

`io.trino.plugin.jdbc.StandardColumnMappings#timestampReadFunction` and 
`io.trino.plugin.jdbc.StandardColumnMappings#toPrestoTimestamp` (BTW let's rename this separately)

-- this is where we do LocalDateTime-based read function. It almost works for your case except it is currently limited to short timestamps, but you know can lift that limitation

Reply:
I moved everything to StandardColumnMappings. For short timestamps, it made sense to round, but for long timestamps, we have to use floor since the nanosecond part contains the remainder. Tests caught it.